app/main.py

```diff
@@ -1,6 +1,22 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    store = {}
+
+    @wraps(func)
+    def wrapped(*args, **kwargs) -> Any:
+
+        key = (func.__name__, args, frozenset(kwargs.items()))
+
+        if key in store:
+            print("Getting from cache")
+            return store[key]
+
+        print("Calculating new result")
+        result = func(*args, **kwargs)
+        store[key] = result
+        return result
```

Comment:
And then because you will have two returns of `store[key]`, change logic to return `store[key]` only once